mp/src/game/client/momentum/ui/leaderboards/LeaderboardsTimes.cpp

```diff
@@ -358,34 +361,33 @@ void CLeaderboardsTimes::LoadLocalTimes(KeyValues* kv)
         V_FixSlashes(path);
 
         FileFindHandle_t found;
+        CUtlStringList filenames;
```

Comment:
I think this entire block of code of finding the filenames and adding it to a CUtlStringList should move into the AsyncLoadReplayFileMultiple method.

Mainly because there needs to be a key into this system so it knows to not step on top of itself. What I'm thinking instead is you pass the mapname in as the first parameter, and in an internal CUtlDict, tie that map name to the request handle. If the handle exists, don't create the new Async request. If it doesn't, do this file searching and then the request.

Reply:
If getting all replays for a single map is really the only use case we need to worry about then I'll add the map name as the first parameter and rename the method accordingly. 

If however this should be a bit more general, maybe the parameter could also be one string the caller constructs that then uses the wildcard-functionality of FindFirstEx in the factory.

Also, could you maybe elaborate a bit more about what you mean with stepping on top of itself? Do you mean the leaderboards generating requests multiple times? I'm not sure I get exactly what you are saying.

Reply:
I think I see what you mean here. The use case is going to be loading a specific file (map finished panel, run submitter, leaderboards on the `run_save` event) and loading all replays for a map (leaderboards Local tab).

In this case, being able to have a `LoadReplaysForMap(mapName, . . .)` that funnels into a `LoadReplaysByFilenameAsync(filenames, . . .)` would work because then the single-file use case can just call that with the one replay path it knows it has.

Reply:
As for stepping on toes, yeah it would just be a matter of preventing the exact same request being done, though I suppose this is a very explicit API so maybe it would be better to debug if it gets called multiple times in the place it's used. So don't worry about that.

Some future expansion could be storing the replays in memory via like a cache in this system, so if a player reloads a map it'll just fetch from cache again instead of doing the requests. Same for individual replay files.